client/src/components/DataEntry/Ballot.tsx

```diff
@@ -73,14 +140,91 @@ const Ballot: React.FC<IProps> = ({
   )
   const ballot = ballots[ballotIx]
 
-  const [auditing, setAuditing] = useState(true)
   const [interpretations, setInterpretations] = useState<
     IBallotInterpretation[]
   >(contests.map(emptyInterpretation))
+  const { confirm, confirmProps } = useConfirm()
+
+  const interpretationsRef = useRef<IBallotInterpretation[]>(
+    contests.map(emptyInterpretation)
+  )
+  interpretationsRef.current = interpretations
+
+  const renderInterpretation = (
+    { interpretation, choiceIds }: IBallotInterpretation,
+    contest: IContest
+  ) => {
+    if (!interpretation) return <div />
+    switch (interpretation) {
+      case Interpretation.VOTE:
+        return contest.choices.map(choice =>
+          choiceIds.includes(choice.id) ? (
+            <h3 key={choice.id}>{choice.name}</h3>
+          ) : null
+        )
+      case Interpretation.BLANK:
+        return <h3>Blank vote</h3>
+      case Interpretation.CONTEST_NOT_ON_BALLOT:
+        return <h3>Not on Ballot</h3>
+      // case Interpretation.CANT_AGREE: (we do not support this case now)
+      default:
+        return null
+    }
+  }
+
+  const setInterpretationsFunc = (
+    newInterpretations: IBallotInterpretation[]
+  ) => {
+    setInterpretations(newInterpretations)
+  }
+
+  const confirmSelections = () => {
+    confirm({
+      title: 'Confirm the Ballot Selections',
+      description: (
+        <>
+          {contests.map((contest, i) => (
+            <div key={contest.id}>
+              <p>{contest.name}</p>
+              {renderInterpretation(interpretationsRef.current[i], contest)}
+              <p>
+                {interpretationsRef.current[i].comment &&
+                  `Comment: ${interpretationsRef.current[i].comment}`}
+              </p>
+            </div>
+          ))}
+        </>
+      ),
+      onYesClick: async () => {
+        submitBallot(
+          ballot.id,
+          BallotStatus.AUDITED,
+          interpretationsRef.current.filter(
+            ({ interpretation }) => interpretation !== null
+          )
+        )
+        nextBallot()
+      },
+      yesButtonLabel: 'Confirm Selections',
+      noButtonLabel: 'Change Selections',
+    })
+  }
 
   const submitNotFound = async () => {
-    await submitBallot(ballot.id, BallotStatus.NOT_FOUND, [])
-    nextBallot()
+    confirm({
+      title: 'Confirm the Ballot Selections',
+      description: (
+        <div>
+          <h3>Ballot Not Found</h3>
+        </div>
+      ),
+      onYesClick: async () => {
+        submitBallot(ballot.id, BallotStatus.NOT_FOUND, [])
```

Comment:
should we `await` this call to `submitBallot` to make sure it succeeds before moving to the next ballot?